semgrep-core/src/core/ast/AST_generic.ml

```diff
@@ -790,11 +790,23 @@ and xml_body =
 (* brackets can be fake '()' for OCaml/Ruby *)
 and arguments = argument list bracket
 
+(*
+   ArgRequired = argument must exist in both pattern and target.
+   ArgOptional = still matches if the argument exists in the target but
```

Comment:
maybe a comment saying that this is currently used just for Bash and Dockerfile, and maybe explain why we need that with an example.

Reply:
I would have preferred a separate ArgKwdOptional maybe.

Reply:
A dedicated ArgKwdOptional also seemed best to me initially but it seemed to make the pattern-matching implementation trickier. Let me see if I can make this work.

Reply:
ok, I changed it, it works fine after all. See upcoming PR.